crypto/objects/o_names.c

```diff
@@ -391,9 +391,9 @@ void OBJ_NAME_cleanup(int type)
     if (type < 0) {
         lh_OBJ_NAME_free(names_lh);
         sk_NAME_FUNCS_pop_free(name_funcs_stack, name_funcs_free);
-        CRYPTO_THREAD_lock_free(obj_lock);
         names_lh = NULL;
         name_funcs_stack = NULL;
+        CRYPTO_THREAD_lock_free(obj_lock);
```

Comment:
Is there still a potential race condition here?

Reply:
You mean the `obj_lock = NULL`?

Reply:
This whole function is a race condition.  It's deleting stuff without holding the lock.  It's freeing the lock so, I'm guessing, the assumption is that it won't run multi-threaded.

Reply:
Yeah I mean what did moving the line down here actually do?

Reply:
I think this is best omitted and marked as a false positive.